### PR TITLE
Feature/oriented-cubes-and-cuboids

### PR DESCRIPTION
**Pull request by mogres**
Problem
=======
Solves #13 

Solution
========
Updated the collision detection for the jitter algorithm, which uses the radial distance from the cube center to detect collisions.

with @meganrm @prismofeverything 

## Type of change
* New feature (non-breaking change which adds functionality)

Change summary:
---------------
* used the collision detection from the spheretree algorithm to detect collision using radial distance from cube center

Steps to Verify:
----------------
run `analyze -p jitter -r cellpack/test-recipes/oriented_cube.json -o out` for oriented cube packing
run `analyze -p jitter -r cellpack/test-recipes/oriented_cuboid.json -o out` for oriented cuboid packing

Screenshots:
-----------------------
Oriented cube packing:
image

Oriented cuboid packing:
image

Keyfiles:
-----------------------
1. single_cube.py
2. oriented_cube.json
3. oriented_cuboid.json

### CODE REVIEW

File: cellpack/autopack/Compartment.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
 import trimesh
 import trimesh.voxel as Voxel
 from scipy import spatial
-from cellpack.autopack.upy.dejavuTk.dejavuHelper import dejavuHelper
+# from cellpack.autopack.upy.dejavuTk.dejavuHelper import dejavuHelper
```
Comment (meganrm) [inline cellpack/autopack/Compartment.py:68]: you can delete this

File: cellpack/autopack/IOutils.py [hunk 2]
```diff
@@ -1,2292 +1,2308 @@
...
+    MultiSphereIngr,
+    SingleCubeIngr,
+    MultiCylindersIngr,
+    GrowIngredient,
+)
+
+encoder.FLOAT_REPR = lambda o: format(o, ".8g")
+
+
+class NumpyArrayEncoder(JSONEncoder):
```
Comment (meganrm) [inline cellpack/autopack/IOutils.py:42]: is this one of the only changes in this file? I wonder why the diff got so messed up

File: cellpack/autopack/ingredient/Ingredient.py [hunk 3]
```diff
@@ -51,7 +51,10 @@
 import collada
 from scipy.spatial.transform import Rotation as R
 from math import sqrt, pi
-from cellpack.autopack.upy.dejavuTk.dejavuHelper import dejavuHelper
+
+from cellpack.autopack.upy.simularium.simularium_helper import simulariumHelper
+# from cellpack.autopack.upy.dejavuTk.dejavuHelper import dejavuHelper
```
Comment (meganrm) [inline cellpack/autopack/ingredient/Ingredient.py:56]: delete this commented out import

File: cellpack/autopack/ingredient/agent.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
+
+
```
Comment (meganrm) [inline cellpack/autopack/ingredient/agent.py:2]: this needs to import numpy

File: cellpack/autopack/IOutils.py (hunk 2 above)
Comment (mogres) [reply cellpack/autopack/IOutils.py:42]: The only change here was removing line 435 and line 436. Not sure why the diff is so weird

File: cellpack/autopack/plotly_result.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def add_ingredient_positions(self, env):
                 if ingr.modelType == "Spheres":
                     self.add_circle(ingr.encapsulatingRadius, pos, ingr.color)
                 elif ingr.modelType == "Cube":
-                    self.add_square(ingr.encapsulatingRadius, pos, ingr.color)
+                    self.add_square(ingr.radii, pos, ingr.color)
```
Comment (meganrm) [inline cellpack/autopack/plotly_result.py:67]: 👍🏻

**Review by meganrm — APPROVED** [review]

**Comment by mogres** [comment]
> Looks good to me, but what is the `google_archive` submodule

Didn't realize that got in there somehow. I'll make a commit to delete it.

**Pull request merged**